### ISSUE TITLE
How to do custom DNS servers over HTTP proxy 

### ISSUE THREAD

**Issue by ghost** [tinyproxy/tinyproxy#483]
## Tinyproxy version
Latest

## Issue

HTTP proxy mode bypasses DNS over HTTPS, and DNS over TLS, so I would expect the HTTP proxy to follow the specified system nameservers, but it DOESN'T.

Please therefore provide instructions on how I can change the DNS resolver that tinyproxy uses as I require custom DNS for my setup to work correctly.

It currently uses some foreign resolver that I've never even seen before, and I want this changed to use my system resolv.conf resolvers.

**Comment by rofl0r** [comment]
in case an upstream is used, tinyproxy just forwards the request to that server, to the DNS lookup is done remotely.
if no upstream is used, DNS resolution is done via the usual getaddrinfo() libc function which typically uses DNS servers from resolv.conf. if this isnt the case on your system, then it's either patched or configured to use some different resolver (e.g. NSS).

**Comment by 20NE** [comment]
tinyproxy 1.11.1

Any progress?
It seems tinyproxy just forwards the request to upstream server, even which is socks4 nor socks5 proxy.

**Comment by rofl0r** [comment]
> It seems tinyproxy just forwards the request to upstream server

correct

> even which is socks4 nor socks5 proxy

the "nor" here doesn't make any sense. what do you want to express ?

**Comment by 20NE** [comment]
Forgive my broken english.
tinyproxy just forwards dns request to upstream server, even the upstream server is socks4 proxy

**Comment by rofl0r** [comment]
yes, and SOCKS4a supports DNS names. SOCKS4a is a requirement for using upstream. i also doubt there are still SOCKS4 servers on the net that dont support SOCKS4a, because it only works with raw ipv4.
